app/provider/session_provider.py

```diff
@@ -0,0 +1,30 @@
+from typing import Optional
+
+import aiohttp
+
+
+class SessionProvider:
```

Comment:
Nit -- some comments here would be helpful

Reply:
Good call! Added some comments to this class.